### ISSUE TITLE
[Accessibility]: History panel view: focus history panel on open

### ISSUE THREAD

**Issue by lrhacker** [sourcegraph/sourcegraph#44583]
### Audit type

Keyboard navigation

### Problem description

#44452 provided screen reader support to properly announce changes after the History panel is expanded, but we still need to do more work to move the focus to the history panel after it has opened. There didn't appear to be a trivial way to accomplish this, so that portion of the work was split into a separate issue.

Also, per @felixfbecker:

> We may be able to address [this] as part of global keyboard shortcuts to jump to different areas too, as those will need the underlying infrastructure to move focus to different areas like the reference panel.

### Expected behavior

After the History panel has been expanded, the focus should move to that section.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
